File: SDKs/amazonaws.com/appmesh/2019-01-25/java/lib/src/main/java/openapisdk/models/shared/VirtualRouterListener.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * VirtualRouterListener
 * An object that represents a virtual router listener.
**/
public class VirtualRouterListener {
    @JsonProperty("portMapping")
    public PortMapping portMapping;
    public VirtualRouterListener withPortMapping(PortMapping portMapping) {
        this.portMapping = portMapping;
        return this;
    }
}

File: SDKs/amazonaws.com/appmesh/2019-01-25/python/src/sdk/models/shared/virtualrouterlistener.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import portmapping as shared_portmapping


@dataclass_json
@dataclasses.dataclass
class VirtualRouterListener:
    r"""VirtualRouterListener
    An object that represents a virtual router listener.
    """
    
    port_mapping: shared_portmapping.PortMapping = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('portMapping') }})
    
